back/app/services/user_role_service.rb

```diff
@@ -1,22 +1,26 @@
 class UserRoleService
```

Comment:
This to me smells, looks and feels like a policy class like we have many of them already: Given an action, a user and an entity, it will tell us if the user can execute the action for that entity.

As someone who did not have a deep dive into our permission system yet, I would expect this to be called `ModeratorPolicy`, because this is what it seems to do?